### PR TITLE
feat(cli): add plan limit recovery guidance

### PR DESCRIPTION
**Pull request by gregory-boch-prisma**
## Summary

- Translate only the structured Management API `error.code === "planLimitReached"` discriminator into stable CLI error `PLAN_LIMIT_REACHED`.
- Enrich that failure with the known workspace ID and one SDK call to `GET /v1/workspaces/{id}/subscription` for `planName`, `usageBlocked`, and the canonical `upgradeUrl`.
- Preserve the standard human stderr / JSON stdout / exit-code pipeline, with safe null metadata and generic Console guidance when enrichment fails.
- Update the Management API SDK to `^1.55.0`, document the public contract, and add deterministic human, JSON, fallback, cancellation, optional-context, and false-positive coverage.

The released SDK types the database failure envelope as `{ error: { code: string, message: string, hint?: string } }`. This change matches only the canonical `planLimitReached` value in that structured `code` field. The current SDK does not expose typed blocked-feature metadata, so `blockedFeature` is explicitly `null` rather than inferred.

### Human output

| Before | After |
| --- | --- |
| Before human terminal output | After human terminal output |

### `--json` output

| Before | After |
| --- | --- |
| Before JSON terminal output | After JSON terminal output |

All evidence uses the same synthetic `ws_synthetic` / `db_synthetic` fixture. No tokens, connection strings, customer identifiers, or billing data are shown.

## Code walkthrough

- `packages/cli/src/lib/database/provider.ts` recognizes the canonical discriminator at the existing database API error translation boundary, before status-specific translations.
- The provider receives already-resolved workspace context and performs at most one cancellation-aware subscription read only after the structured discriminator matches.
- One `CliError` carries the stable code, recovery metadata, exit code `1`, and human lines through the existing command runner and output serializers.
- Lookup failure or partial metadata keeps `PLAN_LIMIT_REACHED`, normalizes unavailable fields to null, and prints generic Console guidance; parameterized tests protect timeout, `503`, `429`, auth, spend-limit, generic API, and `usageBlocked` false positives.
- Public error/output docs now define the code, metadata, stream behavior, and agent/CI branching rule.
- **Why this is the smallest change:** it reuses the existing `databaseApiError` → `CliError` → command output path and avoids a parallel error framework, service, cache, agent subsystem, or Console URL builder.

## How to test

Final results on this branch:

```text
pnpm format
PASS — 200 files formatted; no fixes required in the final run

pnpm lint
PASS — exit 0; repository's existing advisory diagnostics only

pnpm --recursive exec tsc --noEmit
PASS

pnpm --filter @prisma/cli exec vitest run tests/database-plan-limit.test.ts
PASS — 1 file, 18 tests

pnpm --filter @prisma/cli test
PASS — 49 files, 691 tests

pnpm --filter @prisma/cli build
PASS — 104 files built
```

### Manual scenario

1. Use the shared synthetic workspace `<synthetic-workspace-id>` with a database `<synthetic-blocked-database-id>` returning structured `error.code: "planLimitReached"`; keep its token and connection details out of terminal capture.
2. Run `pnpm prisma database show <synthetic-blocked-database-id>` and `pnpm prisma database show <synthetic-blocked-database-id> --json`.
3. Verify both commands exit `1`; human mode writes the plan-limit diagnosis, explicit not-outage wording, current plan, and direct link to stderr.
4. Verify JSON mode writes exactly one parseable failure envelope to stdout with `workspaceId`, `blockedFeature`, `planName`, `usageBlocked`, and `upgradeUrl`, with no human prose on stdout.
5. Open the printed canonical URL and verify it reaches plan selection for `<synthetic-workspace-id>`.
6. Repeat with a stubbed subscription lookup failure, then synthetic timeout, `503`, `429`, auth, spend-limit, generic API, and `usageBlocked: true` responses; verify null-link fallback for the former and no plan-limit false positives for the latter.

## `$code-review`

Reviewed the final `origin/main...HEAD` range with the exact Ignite `$code-review` skill.

- Critical issues: **0**
- Recommendations: **0**
- Final result: **clean**
- Affected checks rerun after review hardening: format, focused recovery tests, recursive type checks, lint, full CLI tests, and build — all pass.

## Out of scope

- No Prisma skills changes.
- No browser UI or automatic checkout/open-browser behavior.
- No inference from status codes, message substrings, simultaneous failures, retry count, or `usageBlocked` alone.

### CODE REVIEW

**Review by luanvdw — COMMENTED** [review]
@gregory-boch-prisma nice work!

One thing I’d address before merge: databaseApiError already knows it should return PLAN_LIMIT_REACHED, but subscription enrichment uses only process-level cancellation. Since the SDK doesn’t impose a request timeout, a stalled lookup could delay the actionable error until the network times out. Could we give enrichment a short deadline, around three seconds, matching the existing update-check timeout, and add a stalled-request test? An enrichment timeout should fall back to null metadata, while caller cancellation should still produce COMMAND_CANCELED.

One non-blocking follow-up: the precedence test covers showDatabase with 404, but the implementation also adds guards for backup 422 and restore 409/404. Could we extend the test table to cover those paths so the structured discriminator remains authoritative?

File: packages/cli/tests/database-plan-limit.test.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+    const error = await provider
+      .showDatabase("db_synthetic")
+      .catch((caught: unknown) => caught);
+
+    expect(error).toBeInstanceOf(CliError);
+    expect((error as CliError).code).toBe(code ?? "DATABASE_API_ERROR");
+    expect((error as CliError).code).not.toBe("PLAN_LIMIT_REACHED");
+    expect(client.GET).toHaveBeenCalledTimes(1);
+  });
```
[the remark this answers is not in the document]
Comment (gregory-boch-prisma) [reply packages/cli/tests/database-plan-limit.test.ts:357]: Confirmed and addressed at the test-contract level in 4a2e33a: the false-positive table no longer asserts `rateLimitReached` or `spendLimitReached` as stable CLI codes. The runtime pass-through predates this PR and is intentionally preserved because this task explicitly requires rate-limit and spend-limit errors to retain their existing behavior; normalizing them here would be an unrelated breaking change.

**Comment by gregory-boch-prisma** [comment]
Addressed the review feedback in 1281457: subscription enrichment now has a 3-second abort deadline and safely falls back to null metadata on timeout; caller cancellation is verified through the CLI pipeline as COMMAND_CANCELED (exit 130). I also added discriminator-precedence coverage for backup 422 and restore 409/404. Focused tests pass 18/18, full CLI tests pass 691/691, and format, lint, recursive types, build, and the required origin/main...HEAD code review are clean.
